### ISSUE TITLE
Camera doesn't work

### ISSUE THREAD

**Issue by natekc** [blumonks/android_device_blu_p6601#1]
```
01-02 13:21:44.483   287   379 D PQ      : [PQ_SERIVCE] DISP_IOCTL_SET_PQ_CAM_PARAM 0...
01-02 13:21:44.538   295   295 I cameraserver: ServiceManager: 0xb6219480
01-02 13:21:44.538   295   295 I CameraService: CameraService started (pid=295)
01-02 13:21:44.538   295   295 I CameraService: CameraService process starting
01-02 13:21:45.150   295   295 I cameraserver: type=1400 audit(0.0:8): avc: denied { write } for name="trace_marker" dev="debugfs" ino=3085 scontext=u:r:cameraserver:s0 tcontext=u:object_r:debugfs:s0 tclass=file permissive=1
01-02 13:21:45.150   295   295 I cameraserver: type=1400 audit(0.0:9): avc: denied { open } for path="/sys/kernel/debug/tracing/trace_marker" dev="debugfs" ino=3085 scontext=u:r:cameraserver:s0 tcontext=u:object_r:debugfs:s0 tclass=file permissive=1
01-02 13:21:45.167   295   295 D ImgSensorDrv: [impSearchSensor]SENSOR search start 
01-02 13:21:45.167   295   295 D ImgSensorDrv: [impSearchSensor]impSearchSensor search to sub
01-02 13:21:45.167   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =10000
01-02 13:21:45.256   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:21:45.256   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:21:45.256   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:21:45.256   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[1][1] 
01-02 13:21:45.256   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:21:45.256   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =10001
01-02 13:21:45.335   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:21:45.335   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:21:45.335   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:21:45.335   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[1][1] 
01-02 13:21:45.335   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:21:45.335   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =10002
01-02 13:21:45.451   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:21:45.451   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:21:45.451   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:21:45.451   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[1][1] 
01-02 13:21:45.451   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:21:45.451   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =10003
01-02 13:21:45.569   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:21:45.569   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:21:45.569   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:21:45.569   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[1][1] 
01-02 13:21:45.569   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:21:45.569   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =10004
01-02 13:21:45.688   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:21:45.688   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:21:45.688   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[1][1] 
01-02 13:21:45.688   295   295 D ImgSensorDrv: [impSearchSensor]found <0x4088/s5k4h8mipiraw_darling/SENSOR_DRVNAME_S5K4H8_MIPI_RAW>
01-02 13:21:45.688   295   295 D ImgSensorDrv: [impSearchSensor]MAIN sensor found:[4]/[0x10004]/[0]/[1] 
01-02 13:21:45.688   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =10005
01-02 13:21:53.787   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:21:53.787   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:21:53.787   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:21:53.788   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[1][1] 
01-02 13:21:53.788   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:21:53.788   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =10006
01-02 13:22:01.847   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:22:01.847   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:22:01.848   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:22:01.848   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[1][1] 
01-02 13:22:01.848   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:22:01.848   295   295 D ImgSensorDrv: [impSearchSensor]m_pstSensorInitFunc[i].getCameraDefault is NULL: 7 
01-02 13:22:01.848   295   295 D ImgSensorDrv: [impSearchSensor]impSearchSensor search to sub
01-02 13:22:01.848   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =20000
01-02 13:22:01.915   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:22:01.915   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:22:01.915   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:22:01.915   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[2][-1] 
01-02 13:22:01.915   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:22:01.915   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =20001
01-02 13:22:01.983   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:22:01.983   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:22:01.983   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:22:01.983   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[2][-1] 
01-02 13:22:01.983   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:22:01.983   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =20002
01-02 13:22:02.049   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:22:02.049   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:22:02.049   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:22:02.049   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[2][-1] 
01-02 13:22:02.049   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:22:02.049   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =20003
01-02 13:22:02.116   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:22:02.116   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:22:02.116   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:22:02.116   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[2][-1] 
01-02 13:22:02.116   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:22:02.116   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =20004
01-02 13:22:02.179   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:22:02.180   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:22:02.180   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:22:02.180   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[2][-1] 
01-02 13:22:02.180   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:22:02.180   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =20005
01-02 13:22:02.261   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:22:02.261   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:22:02.261   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[2][-1] 
01-02 13:22:02.261   295   295 D ImgSensorDrv: [impSearchSensor]found <0x5670/ov5670mipiraw_sunwin/SENSOR_DRVNAME_OV5670_MIPI_RAW>
01-02 13:22:02.261   295   295 D ImgSensorDrv: [impSearchSensor]SUB sensor found:[5]/[0x20005]/[0]/[-1] 
01-02 13:22:02.261   295   295 D ImgSensorDrv: [impSearchSensor]set sensor driver id =20006
01-02 13:22:02.337   295   295 D ImgSensorDrv: [impSearchSensor][impSearchSensor] Err-ctrlCode (I/O error) 
01-02 13:22:02.337   295   295 D ImgSensorDrv: [getCurrentSensorType][getCurrentSensorType] 
01-02 13:22:02.337   295   295 D ImgSensorDrv: [getInfo][getInfo],m_fdSensor = 0x9 
01-02 13:22:02.337   295   295 D ImgSensorDrv: [getSocketPosition][getSocketPosition]:[2][-1] 
01-02 13:22:02.337   295   295 D ImgSensorDrv: [impSearchSensor]sensor ID mismatch
01-02 13:22:02.337   295   295 D ImgSensorDrv: [impSearchSensor]m_pstSensorInitFunc[i].getCameraDefault is NULL: 7 
01-02 13:22:02.338   295   295 D ImgSensorDrv: [impSearchSensor]SENSOR search end: 0x3 /[0x4088][4]/[0xffffff][255]/[0x5670][5]
01-02 13:22:02.338   295   295 D ImgSensorDrv: [init]Warning(  442):sizeof(MUINT16):2,sizeof(MUINT32):4, sizeof(UCHAR):1, sizeof(MUINT64):8sizeof(MUINTPTR):4, sizeof(MUINTPTR *):4, sizeof(MUINT32 *):4, sizeof(UCHAR *):4
01-02 13:22:02.338   295   295 D ImgSensorDrv: [init][init] mUsers = 0
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getResolution][getResolution] ACDK_SENSOR_RESOLUTION_INFO_STRUCT
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getResolution]Warning( 2481):pSensorResolution[0]:0xae22fc00, [1]:0xae22fc6c
01-02 13:22:02.339   295   295 D ImgSensorDrv: [init][featureControl]: SENSOR_FEATURE_GET_PIXEL_CLOCK_FREQ
01-02 13:22:02.339   295   295 E ImgSensorDrv: [init]Err(  520):[init]:  SENSOR_FEATURE_GET_PIXEL_CLOCK_FREQ error = 0
01-02 13:22:02.339   295   295 D ImgSensorDrv: [init][init]: m_LineTimeInus[0] = 31
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getInfo2]prv w=0x660,h=0x4c8
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getInfo2]cap w=0xcc0,h=0x990
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getInfo2]vd  w=0x660,h=0x4c8
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getInfo2]pre GrapX=0x0,GrapY=0x0
01-02 13:22:02.339   295   295 D ImgSensorDrv: [uninit][uninit]imgsensor_drv mUsers = 1
01-02 13:22:02.339   295   295 D ImgSensorDrv: [init]Warning(  442):sizeof(MUINT16):2,sizeof(MUINT32):4, sizeof(UCHAR):1, sizeof(MUINT64):8sizeof(MUINTPTR):4, sizeof(MUINTPTR *):4, sizeof(MUINT32 *):4, sizeof(UCHAR *):4
01-02 13:22:02.339   295   295 D ImgSensorDrv: [init][init] mUsers = 0
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getResolution][getResolution] ACDK_SENSOR_RESOLUTION_INFO_STRUCT
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getResolution]Warning( 2481):pSensorResolution[0]:0xae22fc00, [1]:0xae22fc6c
01-02 13:22:02.339   295   295 E ImgSensorDrv: [init]Err(  553):[init]:  SENSOR_FEATURE_GET_PIXEL_CLOCK_FREQ error = 0
01-02 13:22:02.339   295   295 D ImgSensorDrv: [init][init]: m_LineTimeInus[1] = 31
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getInfo2]prv w=0x500,h=0x3c0
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getInfo2]cap w=0xa00,h=0x780
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getInfo2]vd  w=0x500,h=0x3c0
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getInfo2]pre GrapX=0x1,GrapY=0x0
01-02 13:22:02.339   295   295 D ImgSensorDrv: [uninit][uninit]imgsensor_drv mUsers = 1
01-02 13:22:02.339   295   295 D ImgSensorDrv: [init]Warning(  442):sizeof(MUINT16):2,sizeof(MUINT32):4, sizeof(UCHAR):1, sizeof(MUINT64):8sizeof(MUINTPTR):4, sizeof(MUINTPTR *):4, sizeof(MUINT32 *):4, sizeof(UCHAR *):4
01-02 13:22:02.339   295   295 D ImgSensorDrv: [init][init] mUsers = 0
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getResolution][getResolution] ACDK_SENSOR_RESOLUTION_INFO_STRUCT
01-02 13:22:02.339   295   295 D ImgSensorDrv: [getResolution]Warning( 2481):pSensorResolution[0]:0xae22fc00, [1]:0xae22fc6c
01-02 13:22:02.339   295   295 D ImgSensorDrv: [init][featureControl]: SENSOR_FEATURE_GET_PIXEL_CLOCK_FREQ
01-02 13:22:02.340   295   295 E ImgSensorDrv: [init]Err(  520):[init]:  SENSOR_FEATURE_GET_PIXEL_CLOCK_FREQ error = 0
01-02 13:22:02.340   295   295 D ImgSensorDrv: [init][init]: m_LineTimeInus[0] = 31
01-02 13:22:02.340   295   295 D ImgSensorDrv: [uninit][uninit]imgsensor_drv mUsers = 1
01-02 13:22:02.340   295   295 D ImgSensorDrv: [init]Warning(  442):sizeof(MUINT16):2,sizeof(MUINT32):4, sizeof(UCHAR):1, sizeof(MUINT64):8sizeof(MUINTPTR):4, sizeof(MUINTPTR *):4, sizeof(MUINT32 *):4, sizeof(UCHAR *):4
01-02 13:22:02.340   295   295 D ImgSensorDrv: [init][init] mUsers = 0
01-02 13:22:02.340   295   295 D ImgSensorDrv: [getResolution][getResolution] ACDK_SENSOR_RESOLUTION_INFO_STRUCT
01-02 13:22:02.340   295   295 D ImgSensorDrv: [getResolution]Warning( 2481):pSensorResolution[0]:0xae22fc00, [1]:0xae22fc6c
01-02 13:22:02.340   295   295 E ImgSensorDrv: [init]Err(  553):[init]:  SENSOR_FEATURE_GET_PIXEL_CLOCK_FREQ error = 0
01-02 13:22:02.340   295   295 D ImgSensorDrv: [init][init]: m_LineTimeInus[1] = 31
01-02 13:22:02.340   295   295 D ImgSensorDrv: [uninit][uninit]imgsensor_drv mUsers = 1
01-02 13:22:02.395   295   295 I CameraService: Loaded "MediaTek Camera Module" camera module
01-02 13:22:02.396   295   295 I ServiceManager: Waiting for service media.camera.proxy...
01-02 13:22:03.396   295   295 I ServiceManager: Waiting for service media.camera.proxy...
01-02 13:22:04.396   295   295 I ServiceManager: Waiting for service media.camera.proxy...
01-02 13:22:05.397   295   295 I ServiceManager: Waiting for service media.camera.proxy...
01-02 13:22:08.389   628   628 W PackageManager: Unknown permission com.android.vending.BILLING in package com.sumoing.camu
01-02 13:22:11.216   628   628 I SystemServer: Camera Service
01-02 13:22:11.217   628   628 I SystemServiceManager: Starting com.android.server.camera.CameraService
01-02 13:22:11.229   628   751 I ServiceThread: Enabled StrictMode logging for CameraService_proxy looper.
01-02 13:22:12.700   628   757 W PackageManager: Unknown permission com.android.vending.BILLING in package com.sumoing.camu
01-02 13:22:21.427   815   815 I CameraManagerGlobal: Connecting to camera service
01-02 13:22:21.433   295   729 I CameraService: getCameraCharacteristics: Switching to HAL1 shim implementation...
01-02 13:22:21.433   295   729 I CameraService: CameraService::connect call (PID -1 "cameraserver", camera ID 0) for HAL version default and Camera API version 1
01-02 13:22:21.435   295   729 E CameraService: getCameraPriorityFromProcState: Received invalid process state -1 from ActivityManagerService!
01-02 13:22:21.498   295   729 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=0
01-02 13:22:21.499   295   729 I CameraFlashlight: Opening camera 0
01-02 13:22:21.535   295   729 E CameraFlashlight: Could not open camera 0: -19
01-02 13:22:21.536   295   729 E CameraClient: initialize: Camera 0: unable to initialize device: No such device (-19)
01-02 13:22:21.536   295   729 I CameraFlashlight: Destroying camera 0
01-02 13:22:21.536   295   729 E CameraService: connectHelper: Could not initialize client from HAL module.
01-02 13:22:21.546   295   729 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=1
01-02 13:22:21.547   295   729 I CameraService: disconnect: Disconnected client for camera 0 for PID 295
01-02 13:22:21.547   295   729 E CameraService: initializeShimMetadata: Error initializing shim metadata: Status(-8): '10: connectHelper:946: Failed to initialize camera "0": No such device (-19)'
01-02 13:22:21.559   815   815 E FlashlightController: android.hardware.camera2.CameraAccessException: CAMERA_ERROR (3): connectHelper:946: Failed to initialize camera "0": No such device (-19)
01-02 13:22:21.559   815   815 E FlashlightController: 	at android.hardware.camera2.CameraManager.throwAsPublicException(CameraManager.java:702)
01-02 13:22:21.559   815   815 E FlashlightController: 	at android.hardware.camera2.CameraManager.getOrCreateDeviceIdListLocked(CameraManager.java:778)
01-02 13:22:21.559   815   815 E FlashlightController: 	at android.hardware.camera2.CameraManager.getCameraIdList(CameraManager.java:140)
01-02 13:22:21.559   815   815 E FlashlightController: 	at com.android.systemui.statusbar.policy.FlashlightController.getCameraId(FlashlightController.java:220)
01-02 13:22:21.559   815   815 E FlashlightController: 	at com.android.systemui.statusbar.policy.FlashlightController.tryInitCamera(FlashlightController.java:99)
01-02 13:22:21.559   815   815 E FlashlightController: Caused by: android.os.ServiceSpecificException: connectHelper:946: Failed to initialize camera "0": No such device (-19)
01-02 13:22:21.559   815   815 E FlashlightController: 	at android.hardware.ICameraService$Stub$Proxy.getCameraCharacteristics(ICameraService.java:404)
01-02 13:22:21.559   815   815 E FlashlightController: 	at android.hardware.camera2.CameraManager.getOrCreateDeviceIdListLocked(CameraManager.java:765)
01-02 13:22:22.146   295   295 I CameraService: getCameraCharacteristics: Switching to HAL1 shim implementation...
01-02 13:22:22.147   295   295 I CameraService: CameraService::connect call (PID -1 "cameraserver", camera ID 0) for HAL version default and Camera API version 1
01-02 13:22:22.147   295   295 E CameraService: getCameraPriorityFromProcState: Received invalid process state -1 from ActivityManagerService!
01-02 13:22:22.218   295   295 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=0
01-02 13:22:22.218   295   295 I CameraFlashlight: Opening camera 0
01-02 13:22:22.228   295   295 E CameraFlashlight: Could not open camera 0: -19
01-02 13:22:22.228   295   295 E CameraClient: initialize: Camera 0: unable to initialize device: No such device (-19)
01-02 13:22:22.228   295   295 I CameraFlashlight: Destroying camera 0
01-02 13:22:22.228   295   295 E CameraService: connectHelper: Could not initialize client from HAL module.
01-02 13:22:22.249   295   295 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=1
01-02 13:22:22.250   295   295 I CameraService: disconnect: Disconnected client for camera 0 for PID 295
01-02 13:22:22.250   295   295 E CameraService: initializeShimMetadata: Error initializing shim metadata: Status(-8): '10: connectHelper:946: Failed to initialize camera "0": No such device (-19)'
01-02 13:22:22.256   815   815 E FlashlightController: android.hardware.camera2.CameraAccessException: CAMERA_ERROR (3): connectHelper:946: Failed to initialize camera "0": No such device (-19)
01-02 13:22:22.256   815   815 E FlashlightController: 	at android.hardware.camera2.CameraManager.throwAsPublicException(CameraManager.java:702)
01-02 13:22:22.256   815   815 E FlashlightController: 	at android.hardware.camera2.CameraManager.getOrCreateDeviceIdListLocked(CameraManager.java:778)
01-02 13:22:22.256   815   815 E FlashlightController: 	at android.hardware.camera2.CameraManager.getCameraIdList(CameraManager.java:140)
01-02 13:22:22.256   815   815 E FlashlightController: 	at com.android.systemui.statusbar.policy.FlashlightController.getCameraId(FlashlightController.java:220)
01-02 13:22:22.256   815   815 E FlashlightController: 	at com.android.systemui.statusbar.policy.FlashlightController.tryInitCamera(FlashlightController.java:99)
01-02 13:22:22.256   815   815 E FlashlightController: Caused by: android.os.ServiceSpecificException: connectHelper:946: Failed to initialize camera "0": No such device (-19)
01-02 13:22:22.256   815   815 E FlashlightController: 	at android.hardware.ICameraService$Stub$Proxy.getCameraCharacteristics(ICameraService.java:404)
01-02 13:22:22.256   815   815 E FlashlightController: 	at android.hardware.camera2.CameraManager.getOrCreateDeviceIdListLocked(CameraManager.java:765)
01-02 13:22:22.419   628  1140 I Telecom : CallAudioModeStateMachine: Message received: null.: TS.init->CAMSM.pM_1@AAA_0
01-02 13:23:31.364   628  1322 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.android.camera2/com.android.camera.CameraLauncher bnds=[564,1035][698,1187] (has extras)} from uid 10030 on display 0
01-02 13:23:31.458   628  1125 I ActivityManager: Start proc 1749:com.android.camera2/u0a37 for activity com.android.camera2/com.android.camera.CameraLauncher
01-02 13:23:31.695  1749  1749 V CAM_Profiler: [  0.000ms][ui] GUARD: CameraApp onCreate() - START
01-02 13:23:31.699  1749  1749 V CAM_Profiler: [  0.000ms][ui] GUARD: FirstRunDetector getTimeOfFirstRun - START
01-02 13:23:31.704  1749  1749 V CAM_Profiler: [  4.670ms][ui] GUARD: FirstRunDetector getTimeOfFirstRun - [ 4.670ms] PreferenceManager.getDefaultSharedPreferences
01-02 13:23:31.716  1749  1749 V CAM_Profiler: [ 16.363ms][ui] GUARD: FirstRunDetector getTimeOfFirstRun - [11.692ms] preferences.getLong
01-02 13:23:31.720  1749  1749 I CAM_Profiler: [ 20.746ms][ui] GUARD: FirstRunDetector getTimeOfFirstRun - STOP
01-02 13:23:31.723  1749  1749 I CAM_Profiler: [ 30.169ms][ui] GUARD: CameraApp onCreate() - [30.169ms] initializeTimeOfFirstRun
01-02 13:23:31.728  1749  1749 V CAM_Profiler: [ 34.382ms][ui] GUARD: CameraApp onCreate() - [ 4.212ms] UsageStatistics.initialize
01-02 13:23:31.736  1749  1749 V CAM_Profiler: [ 42.321ms][ui] GUARD: CameraApp onCreate() - [ 7.939ms] clearNotifications
01-02 13:23:31.739  1749  1749 I CAM_Profiler: [ 42.321ms][ui] GUARD: CameraApp onCreate() - STOP
01-02 13:23:31.838  1749  1749 V CAM_QuickActivity: START onCreate: Activity = com.android.camera.CameraActivity@8c1d466
01-02 13:23:31.840  1749  1749 V CAM_Profiler: [  0.000ms][ui] GUARD: CameraActivity.onCreateTasks - START
01-02 13:23:31.841  1749  1749 D CAM_LocationManager: Using legacy location provider.
01-02 13:23:31.916  1749  1749 D CAM_MemoryManagerImpl: Max native memory: 179 MB
01-02 13:23:31.937  1749  1749 I CAM_OneCamFtrCnfgCrtr: CaptureModule? true
01-02 13:23:32.002  1749  1749 I CAM_Profiler: [161.801ms][ui] GUARD: CameraActivity.onCreateTasks - [51.903ms] Glide.setup
01-02 13:23:32.012  1749  1749 V CAM_Profiler: [172.560ms][ui] GUARD: CameraActivity.onCreateTasks - [10.759ms] OneCameraManager.get
01-02 13:23:32.041  1749  1749 I CameraManagerGlobal: Connecting to camera service
01-02 13:23:32.048   295   729 I CameraService: getCameraCharacteristics: Switching to HAL1 shim implementation...
01-02 13:23:32.048   295   729 I CameraService: CameraService::connect call (PID -1 "cameraserver", camera ID 0) for HAL version default and Camera API version 1
01-02 13:23:32.049   295   729 E CameraService: getCameraPriorityFromProcState: Received invalid process state -1 from ActivityManagerService!
01-02 13:23:32.093   295   729 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=0
01-02 13:23:32.094   295   729 I CameraFlashlight: Opening camera 0
01-02 13:23:32.099   295   729 E CameraFlashlight: Could not open camera 0: -19
01-02 13:23:32.099   295   729 E CameraClient: initialize: Camera 0: unable to initialize device: No such device (-19)
01-02 13:23:32.099   295   729 I CameraFlashlight: Destroying camera 0
01-02 13:23:32.099   295   729 E CameraService: connectHelper: Could not initialize client from HAL module.
01-02 13:23:32.106   295   729 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=1
01-02 13:23:32.107   295   729 I CameraService: disconnect: Disconnected client for camera 0 for PID 295
01-02 13:23:32.107   295   729 E CameraService: initializeShimMetadata: Error initializing shim metadata: Status(-8): '10: connectHelper:946: Failed to initialize camera "0": No such device (-19)'
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: Could not get device listing from camera subsystem
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: android.hardware.camera2.CameraAccessException: CAMERA_ERROR (3): connectHelper:946: Failed to initialize camera "0": No such device (-19)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.hardware.camera2.CameraManager.throwAsPublicException(CameraManager.java:702)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.hardware.camera2.CameraManager.getOrCreateDeviceIdListLocked(CameraManager.java:778)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.hardware.camera2.CameraManager.getCameraIdList(CameraManager.java:140)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at com.android.ex.camera2.portability.AndroidCamera2AgentImpl.updateCameraDevices(AndroidCamera2AgentImpl.java:110)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at com.android.ex.camera2.portability.AndroidCamera2AgentImpl.<init>(AndroidCamera2AgentImpl.java:100)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at com.android.ex.camera2.portability.CameraAgentFactory.getAndroidCameraAgent(CameraAgentFactory.java:129)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at com.android.camera.CameraActivity.onCreateTasks(CameraActivity.java:1515)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at com.android.camera.util.QuickActivity.onCreate(QuickActivity.java:114)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.app.Activity.performCreate(Activity.java:6684)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2637)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.app.ActivityThread.-wrap12(ActivityThread.java)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.os.Handler.dispatchMessage(Handler.java:102)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.os.Looper.loop(Looper.java:154)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.app.ActivityThread.main(ActivityThread.java:6186)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at java.lang.reflect.Method.invoke(Native Method)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: Caused by: android.os.ServiceSpecificException: connectHelper:946: Failed to initialize camera "0": No such device (-19)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.os.Parcel.readException(Parcel.java:1698)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.os.Parcel.readException(Parcel.java:1637)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.hardware.ICameraService$Stub$Proxy.getCameraCharacteristics(ICameraService.java:404)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	at android.hardware.camera2.CameraManager.getOrCreateDeviceIdListLocked(CameraManager.java:765)
01-02 13:23:32.118  1749  1749 E CAM2PORT_AndCam2AgntImp: 	... 18 more
01-02 13:23:32.139   295   729 I CameraService: CameraService::connect call (PID -1 "com.android.camera2", camera ID 0) for HAL version default and Camera API version 1
01-02 13:23:32.175   295   729 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=0
01-02 13:23:32.176   295   729 I CameraFlashlight: Opening camera 0
01-02 13:23:32.181   295   729 E CameraFlashlight: Could not open camera 0: -19
01-02 13:23:32.181   295   729 E CameraClient: initialize: Camera 0: unable to initialize device: No such device (-19)
01-02 13:23:32.181   295   729 I CameraFlashlight: Destroying camera 0
01-02 13:23:32.181   295   729 E CameraService: connectHelper: Could not initialize client from HAL module.
01-02 13:23:32.187   295   729 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=1
01-02 13:23:32.189   295   729 I CameraService: disconnect: Disconnected client for camera 0 for PID 1749
01-02 13:23:32.189  1749  1749 W CameraBase: An error occurred while connecting to camera 0: Service not available
01-02 13:23:32.190   295  1052 I CameraService: CameraService::connect call (PID -1 "com.android.camera2", camera ID 1) for HAL version default and Camera API version 1
01-02 13:23:32.219   295  1052 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=1, newStatus=0
01-02 13:23:32.219   295  1052 I CameraFlashlight: Opening camera 1
01-02 13:23:32.224   295  1052 E CameraFlashlight: Could not open camera 1: -19
01-02 13:23:32.224   295  1052 E CameraClient: initialize: Camera 1: unable to initialize device: No such device (-19)
01-02 13:23:32.224   295  1052 I CameraFlashlight: Destroying camera 1
01-02 13:23:32.224   295  1052 E CameraService: connectHelper: Could not initialize client from HAL module.
01-02 13:23:32.230   295  1052 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=1, newStatus=1
01-02 13:23:32.232   295  1052 I CameraService: disconnect: Disconnected client for camera 1 for PID 1749
01-02 13:23:32.233  1749  1749 W CameraBase: An error occurred while connecting to camera 1: Service not available
01-02 13:23:32.236  1749  1749 I CAM_Profiler: [395.654ms][ui] GUARD: CameraActivity.onCreateTasks - [102.130ms] computePictureSizes
01-02 13:23:32.360  1749  1749 W CAM_Log : Tag RoundedThumbnailView is 1 chars longer than limit.
01-02 13:23:32.784  1749  1749 I CAM_Profiler: [944.011ms][ui] GUARD: CameraActivity.onCreateTasks - [532.094ms] setContentView()
01-02 13:23:32.848  1749  1749 I CAM_Profiler: [1008.417ms][ui] GUARD: CameraActivity.onCreateTasks - [64.406ms] Configure Camera UI
01-02 13:23:32.881  1749  1749 V CAM_CameraController: Closing camera
01-02 13:23:32.882  1749  1768 W CAM2PORT_AndCamAgntImp: Releasing camera without any camera opened.
01-02 13:23:32.882  1749  1749 V CAM_ModulesInfo: EnableCaptureModule = true
01-02 13:23:32.892  1749  1749 V CAM_Profiler: [  0.000ms][ui] GUARD: new CaptureModule - START
01-02 13:23:32.901  1749  1749 V CAM_Profiler: [  8.002ms][ui] GUARD: new CaptureModule - STOP
01-02 13:23:32.901  1749  1749 V CAM_SettingsManager: listeners: [com.android.camera.app.CameraAppUI@725bdf]
01-02 13:23:32.921  1749  1749 V CAM_SettingsManager: listeners: [com.android.camera.app.CameraAppUI@725bdf, com.android.camera.ButtonManager@f003318]
01-02 13:23:32.924  1749  1749 V CAM_SettingsManager: listeners: [com.android.camera.app.CameraAppUI@725bdf, com.android.camera.ButtonManager@f003318, com.android.camera.widget.IndicatorIconController@edd7b56]
01-02 13:23:32.927  1749  1749 I CAM_Profiler: [1087.571ms][ui] GUARD: CameraActivity.onCreateTasks - [24.577ms] Init Current Module UI
01-02 13:23:32.929  1749  1749 V CAM_Profiler: [  0.000ms][ui] GUARD: CaptureModule.init - START
01-02 13:23:32.929  1749  1749 D CAM_CaptureModule: init UseAutotransformUiLayout = false
01-02 13:23:32.934  1749  1749 D CAM_Camera2OneCamMgr: Getting First BACK Camera
01-02 13:23:32.935   295   729 I CameraService: getCameraCharacteristics: Switching to HAL1 shim implementation...
01-02 13:23:32.935   295   729 I CameraService: CameraService::connect call (PID -1 "cameraserver", camera ID 0) for HAL version default and Camera API version 1
01-02 13:23:32.936   295   729 E CameraService: getCameraPriorityFromProcState: Received invalid process state -1 from ActivityManagerService!
01-02 13:23:32.971   295   729 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=0
01-02 13:23:32.972   295   729 I CameraFlashlight: Opening camera 0
01-02 13:23:32.978   295   729 E CameraFlashlight: Could not open camera 0: -19
01-02 13:23:32.978   295   729 E CameraClient: initialize: Camera 0: unable to initialize device: No such device (-19)
01-02 13:23:32.978   295   729 I CameraFlashlight: Destroying camera 0
01-02 13:23:32.978   295   729 E CameraService: connectHelper: Could not initialize client from HAL module.
01-02 13:23:32.982   295   729 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=1
01-02 13:23:32.984   295   729 I CameraService: disconnect: Disconnected client for camera 0 for PID 295
01-02 13:23:32.984   295   729 E CameraService: initializeShimMetadata: Error initializing shim metadata: Status(-8): '10: connectHelper:946: Failed to initialize camera "0": No such device (-19)'
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: Unable to get camera ID
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: android.hardware.camera2.CameraAccessException: CAMERA_ERROR (3): connectHelper:946: Failed to initialize camera "0": No such device (-19)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.hardware.camera2.CameraManager.throwAsPublicException(CameraManager.java:702)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.hardware.camera2.CameraManager.getOrCreateDeviceIdListLocked(CameraManager.java:778)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.hardware.camera2.CameraManager.getCameraIdList(CameraManager.java:140)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at com.android.camera.one.v2.Camera2OneCameraManagerImpl.findFirstCameraIdFacing(Camera2OneCameraManagerImpl.java:156)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at com.android.camera.one.v2.Camera2OneCameraManagerImpl.findFirstBackCameraId(Camera2OneCameraManagerImpl.java:135)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at com.android.camera.one.v2.Camera2OneCameraManagerImpl.findCameraId(Camera2OneCameraManagerImpl.java:128)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at com.android.camera.one.v2.Camera2OneCameraManagerImpl.findFirstCameraFacing(Camera2OneCameraManagerImpl.java:102)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at com.android.camera.CaptureModule.updateCameraCharacteristics(CaptureModule.java:390)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at com.android.camera.CaptureModule.init(CaptureModule.java:417)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at com.android.camera.CameraActivity.onCreateTasks(CameraActivity.java:1665)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at com.android.camera.util.QuickActivity.onCreate(QuickActivity.java:114)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.app.Activity.performCreate(Activity.java:6684)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2637)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.app.ActivityThread.-wrap12(ActivityThread.java)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.os.Handler.dispatchMessage(Handler.java:102)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.os.Looper.loop(Looper.java:154)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.app.ActivityThread.main(ActivityThread.java:6186)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at java.lang.reflect.Method.invoke(Native Method)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: Caused by: android.os.ServiceSpecificException: connectHelper:946: Failed to initialize camera "0": No such device (-19)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.os.Parcel.readException(Parcel.java:1698)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.os.Parcel.readException(Parcel.java:1637)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.hardware.ICameraService$Stub$Proxy.getCameraCharacteristics(ICameraService.java:404)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	at android.hardware.camera2.CameraManager.getOrCreateDeviceIdListLocked(CameraManager.java:765)
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: 	... 21 more
01-02 13:23:32.987  1749  1749 W CAM_Camera2OneCamMgr: No back-facing camera found.
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: Handling Camera Access Failure:
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: java.lang.Exception
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at com.android.camera.FatalErrorHandlerImpl.onGenericCameraAccessFailure(FatalErrorHandlerImpl.java:84)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at com.android.camera.CaptureModule.updateCameraCharacteristics(CaptureModule.java:396)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at com.android.camera.CaptureModule.init(CaptureModule.java:417)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at com.android.camera.CameraActivity.onCreateTasks(CameraActivity.java:1665)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at com.android.camera.util.QuickActivity.onCreate(QuickActivity.java:114)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at android.app.Activity.performCreate(Activity.java:6684)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2637)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at android.app.ActivityThread.-wrap12(ActivityThread.java)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at android.os.Handler.dispatchMessage(Handler.java:102)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at android.os.Looper.loop(Looper.java:154)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at android.app.ActivityThread.main(ActivityThread.java:6186)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at java.lang.reflect.Method.invoke(Native Method)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
01-02 13:23:32.989  1749  1749 E CAM_FatalErrorHandler: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
01-02 13:23:32.989  1749  1749 E CAM_CameraUtil: Show fatal error dialog
01-02 13:23:33.102  1749  1749 I CAM_Profiler: [1262.615ms][ui] GUARD: CameraActivity.onCreateTasks - [175.044ms] Init CurrentModule
01-02 13:23:33.161  1749  1749 I CAM_Profiler: [1322.660ms][ui] GUARD: CameraActivity.onCreateTasks - STOP
01-02 13:23:33.162  1749  1749 V CAM_QuickActivity: END onCreate: Activity = com.android.camera.CameraActivity@8c1d466
01-02 13:23:33.196  1749  1810 D CAM_Storage: External storage state=mounted
01-02 13:23:33.210  1749  1749 V CAM_QuickActivity: START onStart: Activity = com.android.camera.CameraActivity@8c1d466
01-02 13:23:33.219  1749  1749 V CAM_QuickActivity: END onStart: Activity = com.android.camera.CameraActivity@8c1d466
01-02 13:23:33.225  1749  1749 V CAM_QuickActivity: START onResume: Activity = com.android.camera.CameraActivity@8c1d466
01-02 13:23:33.234  1749  1749 V CAM_QuickActivity: onResume(): isKeyguardLocked() = false
01-02 13:23:33.235  1749  1749 V CAM_QuickActivity: onResume --> onResumeTasks()
01-02 13:23:33.242  1749  1749 V CAM_Profiler: [  0.000ms][ui] GUARD: CameraActivity.resume - START
01-02 13:23:33.242  1749  1749 V
```

**Comment by natekc** [comment]
~~According to vampirefo, the camera libs depend on lib3a.so and libdpframework.so, which somehow makes it incompatible with libc.so and libandroidruntime.so..~~ (this is totally irrelevant)

**Comment by zhaofengli** [comment]
danielhk's [camera wrapper](https://github.com/danielhk/android_device_lenovo_aio_otfp/tree/cm-14.1/camera_wrapper) could be of use. Also see [his changes](https://github.com/danielhk/proprietary_vendor_lenovo_aio_otfp/commit/814c267a00ad5543dc9fe8e567913f484787df84) to the vendor makefiles.

**Comment by natekc** [comment]
Looks like our libs don't crash at the same place that danielhk's did.. 

Somehow we get the wrong PID for the media process when trying to connect to the camera service from cameraManagerGlobal.

**Comment by natekc** [comment]
It really is a mystery why the libs aren't working.  According to DeckerSu who had very similar error (or lack of) message to us, camera can be fixed by adding all the missing libs.  Unfortunately that didn't work.. I used ndk-depends to show all deps for the camera hal.  I even dlopen each individual camera related library and they were all fine.

**Comment by natekc** [comment]
Confirmed that danielhk's camera wrapper would be useless for us.  getResolution() works correctly for our HAL because I can see the calls completing successfully in kernel space.

**Comment by Zombie-Ryushu** [comment]
Has anyone attempted to use a Kernel driver from another similar BLU Device that runs Nougat?

**Comment by natekc** [comment]
No that won't fix anything.  The problem isn't in the kernel.

**Comment by rob-3** [comment]
Just a shot in the dark... mrmazak has improved the stability of his camera on his LineageOS 13 port and Slim6 port by substituting Open Camera instead of the default app. Any chance this would help us here?

**Comment by natekc** [comment]
Sorry, open camera won't help us here if our camera isn't working at all.  The problem is at the library level (probably libcamera_client.so or libcameraservice.so).  My current guess is that MTK modified something there that is needed for the camera to work.

**Comment by Zombie-Ryushu** [comment]
Exactly. The most sensible solution I see is for one of the team members to buy a Nougat MediaTek device, figure out what was done, then build what we learn into all subsequent MediaTek Roms.

**Comment by rob-3** [comment]
Do we just need to get some MediaTek nougat camera libraries from some device?

**Comment by Zombie-Ryushu** [comment]
Some additional new stuff has been learned from the BLU R2, but, as far as I know. The Nougat Rom is still broken with regard to Bluetooth, and the Camera. You can still use Bluetooth and the Camera with the LineageOS 13.1 Marshmellow Rom, but you'll be vulnerable to KRACK

**Comment by Zombie-Ryushu** [comment]
Has anyone worked on this?

**Comment by aaronbooker** [comment]
I would donate my R1 HD for the duration of figuring this out and pay shipping both ways... :-)

I really appreciate what you guys are doing.  Thank you!  Would love to get Lineage OS on it with a working camera.

**Comment by Zombie-Ryushu** [comment]
That exists, the LineageOS 13 Rom has a working Camera. Its LineageOS 14 thats a problem.

**Comment by aaronbooker** [comment]
Sorry - I realized that.  I meant a working camera without security exploits.  In reading the thread, it sounded like a device could be useful for the devs?

**Comment by Zombie-Ryushu** [comment]
Not another R1 HD. The issue is this. BLU Either doesn't have the Drivers, or won't because it threatens BLU's Business model, produce a fully working Nougat Rom for the R1 HD. The WPA Supplicant Krack issue can be fixed in the Marshmellow rom by recompiling WPA Supplicant independently. That would require either a system loaded Zip (Loaded from TWRP) or  a completely new Marshmellow Rom. Remember, KRACK is in WPA Suppplicant. Not the Kernel.

Its very unusual for BLU to update their devices an Android version. When they do,its usually the high end models with Qualcomm chips rarther than MediaTek. This is why they don't sell Phones with Oreo.  

Either they don't have the drivers, or they are holding them back to use as a carrot and stick to get consumers to Purchase new devices. 

What was being was attempted to  get the R1 HD to use Drivers from the BLU R2 LTE.

**Comment by rob-3** [comment]
@aaronbooker Kazudante on XDA only stopped development due to lack of a device. Maybe you could get in touch with him?

**Comment by Zombie-Ryushu** [comment]
I own one. But  am using it. Can't he just buy a used one?

**Comment by Mohammadalin** [comment]
can i change this two file form andoid 5.1 to 7.1 
will it work ?
